File: code/community/1304604967260131333/deepgram-rttm-file-format-diarization.js
const fetch = require('node-fetch');
require('dotenv').config();

const apiKey = process.env.DEEPGRAM_API_KEY;
const audioUrl = 'https://your-audio-file-url.com/audio.wav';

async function transcribeAudio() {
  const response = await fetch('https://api.deepgram.com/v1/listen', {
    method: 'POST',
    headers: {
      'Authorization': `Token ${apiKey}`,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      url: audioUrl,
      diarize: true,
      punctuate: true,
    }),
  });

  const data = await response.json();
  console.log(data);
}

transcribeAudio().catch(console.error);

File: code/community/1304604967260131333/deepgram-rttm-file-format-diarization.py
import os
import requests

# Set the API details
api_key = os.getenv('DEEPGRAM_API_KEY')
audio_url = 'https://your-audio-file-url.com/audio.wav'

# Define the endpoint and headers
url = 'https://api.deepgram.com/v1/listen'
headers = {
    'Authorization': f'Token {api_key}',
    'Content-Type': 'application/json'
}

# Set query parameters
params = {
    'diarize': 'true',
    'punctuate': 'true'
}

# Make the request to the Deepgram API
response = requests.post(url, headers=headers, params=params, data={"url": audio_url})

# Print the response
print(response.json())